### PR TITLE
Print summary at the end of the configure script

### PR DESCRIPTION
**Pull request by dgoulet-tor**
Signed-off-by: David Goulet <dgoulet@torproject.org>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+PPRINT_SET_TS(38)
+
+AS_ECHO
+AS_ECHO("${PPRINT_COLOR_BLDBLU}Tor $PACKAGE_VERSION $PPRINT_COLOR_RST")
+AS_ECHO
+
+PPRINT_SUBTITLE([Build Features])
+
+test "x$enable_gcc_warnings" != "xno" && value=1 || value=0
+PPRINT_PROP_BOOL([GCC Warnings], $value)
```
Comment (nmathewson) [inline configure.ac]: We should call this "verbose warnings"; it applies to clang too.

Comment (nmathewson) [reply configure.ac]: Maybe it would be a good idea to also print, for this setting and all the others, how to override it?

File: configure.ac [hunk 2]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+
+test "x$enable_fatal_warnings" != "xno" && value=1 || value=0
+PPRINT_PROP_BOOL([GCC Fatal Warnings], $value)
```
Comment (nmathewson) [inline configure.ac]: Let's call this "Warnings are fatal"

File: configure.ac [hunk 3]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+
+test "x$enable_coverage" = "xyes" && value=1 || value=0
+PPRINT_PROP_BOOL([Code Coverage], $value)
```
Comment (nmathewson) [inline configure.ac]: This maybe belongs in a testing section.

File: configure.ac [hunk 4]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+fi
+
+if test "x$enable_static_zlib" = "xyes"; then
+  PPRINT_PROP_STRING([zlib1g], [$TOR_LIBDIR_zlib])
+else
+  PPRINT_PROP_BOOL([zlib1g], $value)
+fi
+
+AS_ECHO
+PPRINT_SUBTITLE([Security])
```
Comment (nmathewson) [inline configure.ac]: Maybe we should have an "optional libraries" section for NSS and libscrypt and libsystemd.  This section makes it sould like your system is less secure if it doesn't have NSS, which isn't really true.

File: configure.ac [hunk 5]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+PPRINT_PROP_BOOL([libnss Support], $value)
+
+test "x$enable_seccomp" = "xyes" && value=1 || value=0
+PPRINT_PROP_BOOL([libseccomp Support], $value)
+
+test "x$enable_libscrypt" = "xyes" && value=1 || value=0
+PPRINT_PROP_BOOL([libscrypt support], $value)
+
+test "x$enable_fragile_hardening" != "xno" && value=1 || value=0
+PPRINT_PROP_BOOL([libasan support], $value)
```
Comment (nmathewson) [inline configure.ac]: Let's call this "fragile hardening" to match the option.  Also let's mark it as "not for production use" somehow-- people should use this in production.

File: configure.ac [hunk 6]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+
+test "x$enable_gcc_hardening" = "xyes" && value=1 || value=0
+PPRINT_PROP_BOOL([GCC Hardening], $value)
```
Comment (nmathewson) [inline configure.ac]: Let's call this "compiler hardening"? 

This option is supposed to be on by default, but when I ran the configure script, it said that it was off.  I think there might be a  bug.

File: configure.ac [hunk 7]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+PPRINT_PROP_BOOL([HTML Manual], $value)
+
+AS_ECHO
+PPRINT_SUBTITLE([Tests])
+
+test "x$enable_unittests" != "xno" && value=1 || value=0
+PPRINT_PROP_BOOL([Unit tests], $value)
+
+test "x$enable_asserts_in_tests" = "xyes" && value=1 || value=0
+PPRINT_PROP_BOOL([Unit tests assert()], $value)
```
Comment (nmathewson) [inline configure.ac]: This option was mis-reported when I ran the script.  It is almost always on.

Its name is a little confusing, too. How about invert this, can call it "assert()s disabled for unit tests"

We should also mark this option as not safe to use --disable-asserts-in-tests in production.

File: configure.ac [hunk 8]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+test "x$enable_libfuzzer" = "xyes" && value=1 || value=0
+PPRINT_PROP_BOOL([libFuzzer support], $value)
+
+test "x$enable_oss_fuzz" = "xyes" && value=1 || value=0
+PPRINT_PROP_BOOL([OSS-Fuzz support], $value)
+
+AS_ECHO
+PPRINT_SUBTITLE([Install Directories])
+
+report_mandir="`eval eval echo $mandir`"
```
Comment (nmathewson) [inline configure.ac]: eval eval?  I don't understand this.

File: configure.ac [hunk 9]
```diff
@@ -2519,3 +2519,118 @@ https://trac.torproject.org/projects/tor/wiki/doc/TorFragileHardening
...
+PPRINT_PROP_STRING([Binaries], [$BINDIR])
+PPRINT_PROP_STRING([Configuration], [$CONFDIR])
+PPRINT_PROP_STRING([Man Pages], [$report_mandir])
```
Comment (nmathewson) [inline configure.ac]: Is there any way that we can get the complete configure command line as it was provided to us?  For a lot of cases, that would be more useful for debugging than all the rest of this stuff.

File: configure.ac (hunk 8 above)
Comment (dgoulet-tor) [reply configure.ac]: Because just one eval gives: `${prefix}/share/man` so the second one evaluates that prefix.

File: configure.ac (hunk 1 above)
Comment (dgoulet-tor) [reply configure.ac]: Yup. There is one fixup commit that fixes it all. I recommend you rerun the configure script to see the end result. I changed many things from your review :).

File: configure.ac (hunk 2 above)
Comment (dgoulet-tor) [reply configure.ac]: +1

File: configure.ac (hunk 3 above)
Comment (dgoulet-tor) [reply configure.ac]: +1

File: configure.ac (hunk 4 above)
Comment (dgoulet-tor) [reply configure.ac]: +1

File: configure.ac (hunk 5 above)
Comment (dgoulet-tor) [reply configure.ac]: +1

File: configure.ac (hunk 6 above)
Comment (dgoulet-tor) [reply configure.ac]: +1

I fixed the bug I believe.

File: configure.ac (hunk 7 above)
Comment (dgoulet-tor) [reply configure.ac]: +1

File: configure.ac (hunk 9 above)
Comment (dgoulet-tor) [reply configure.ac]: I added a last line that does that :).

**Pull request closed without merging**